ui/hooks/useTransactionDisplayData.js

```diff
@@ -197,11 +197,12 @@ export function useTransactionDisplayData(transactionGroup) {
         tokenId === transactionDataTokenId,
     );
 
-  const tokenDisplayValue = useTokenDisplayValue(
-    primaryTransaction?.txParams?.data,
-    token,
-    isTokenCategory,
-  );
+  const tokenDisplayValue =
+    useTokenDisplayValue(
+      primaryTransaction?.txParams?.data,
+      token,
+      isTokenCategory,
+    ) ?? transactionDataTokenId;
```

Comment:
It's a little weird to use a token id variable to populate the token value. 

Also, `useTokenDisplayValue` performs a calculation `shouldCalculateTokenValue` where it intentionally returns null sometimes, which this `??` may overwrite with `transactionDataTokenId` in some cases it shouldn't.

I assume this PR works because `transactionDataTokenId` falls back to `getTokenValueParam` above.  But then the fix seems to belong in `useTokenDisplayValue` instad.  Something like this: https://github.com/MetaMask/metamask-extension/commit/e60047b2538b01a04aa72db40a4ade0ab0c7e7e0

I'm guessing that because `0` is falsey, zero decimal tokens are causing `shouldCalculateTokenValue` to be false.  But zero decimal tokens are allowed and their value should be calculated.

Reply:
i totally agree , you're right.
just updated